client/light/src/blockchain.rs

```diff
@@ -173,3 +173,21 @@ impl<S, Block: BlockT> RemoteBlockchain<Block> for Blockchain<S>
 		}))
 	}
 }
+
+impl<S: Storage<Block>, Block: BlockT> ProvideChtRoots<Block> for Blockchain<S> {
+	fn header_cht_root(
+		&self,
+		cht_size: NumberFor<Block>,
+		block: NumberFor<Block>,
+	) -> sp_blockchain::Result<Option<Block::Hash>> {
+		self.storage().header_cht_root(cht_size, block)
```

Comment:
Ok, there's still some confusion between comments + full impl + light impl. Full impl would return `Some(_)` for all headers except genesis. And the light client only returns `Some(_)` for headers where CHT is built.

Reply:
I've made the comments clearer.

Reply:
Actually, I'm not sure how it works in the light db either. `sc_client_db::light::get_cht_works` has this line:
```rust
let cht_root_2 = db.header_cht_root(cht_size, cht::start_number(cht_size, 0) + cht_size / 2).unwrap().unwrap();
```
So it doesn't seem to need to be every 2048 blocks.

Reply:
Ok, you must be right :) Sorry - I haven't seen this code for a while.